### DISCUSSION TITLE
I need to export the drainage basin asymmetry output map as SHP

### DISCUSSION THREAD

**Discussion by AllanLopez** [wschwanghart/topotoolbox#35] [Q&A]
Dear Wolfgang and Colleages,
I am very sorry to have confused the location of the following question which you have already answered, and it is working very fine.
I enclose Wolfgang kind solution.

I need to export the drainage basin asymmetry output map as SHP and when I apply: shapewrite(DB,'NAME') I only get the individual basins without the right and left sides of the trunks. I will deeply appreciate if you can modify the shapewrite instruction or code to get the tilted sides and if possible, to identify them with a shadow or ticker outline.  I have unsuccessfully tried several combinations. Thank you very much in advance. Allan Lopez

**Comment by wschwanghart** [comment]
Hi Allan,

Currently, the individual sides are not returned as grids or polygon, but here's some code so that this can be done easily.

```
DEM = GRIDobj('srtm_bigtujunga30m_utm11.tif');
FD  = FLOWobj(DEM);
S   = STREAMobj(FD,'minarea',1000);
[D,AS,ST] = dbasymmetry(FD,S);

% Set D (half catchments) to either 0 or 1 (in the output, the stream has
% the value 0.5).
D.Z = uint32(D.Z>0);
% Compute drainage basins
DB  = drainagebasins(FD,ST);
% Multiply drainage basin index and add D. Hence, there is a unique value
% for every half catchment.
DB  = DB*2 + D;

% Compute polygon
MS  = GRIDobj2polygon(DB);
shapewrite(MS,'half_catchments.shp')**
```

Hope this works.
Best, Wolfgang

**Comment by fredM25** [comment]
Dear Wolfgang and Colleagues,

I need to export the calculated specific basin asymmetry values with their respective drainage basins in excel or tabular form in matlab for further morphometric analyses.
I will appreciate your kind help for this.
Thanks in advance
Fred

**Comment by wschwanghart** [comment]
HI Fred, it'll be helpful to send some code to see where you are stuck. Most likely, you'll will need `zonalstats`.

**Reply by fredM25** [reply]
Hi Wolfgang,

Here is the flow of codes I applied
>> DEM=GRIDobj("output_SRTMGL1.tif");
>> imageschs(DEM)
>> DEM.Z(DEM.Z==767) = nan;
>> DEM = reproject2utm(DEM,30);
>> FD  = FLOWobj(DEM);
>> S=STREAMobj(FD,'minarea',1000);
>> LAKE = isnan(DEM);
>> LAKE = dilate(LAKE,ones(3));
>>  S   = modify(S,'upstreamto',LAKE);
>> A   = flowacc(FD);
>> ST  = trunk(S,A);
>>  [D,AS] = dbasymmetry(FD,ST,'extractlongest',true);
>> imageschs(DEM,AS,'caxis',[0 1], ...
'colormap',ttscm('broc'))
>> hold on
>> plot(ST,'b')
>> DB = drainagebasins(FD,ST);
>>t = table();
>> [t.DB,t.AS] = drainagebasinstats(DB,AS);
Incorrect number or types of inputs or outputs for function drainagebasinstats.

I needed to export the specific values of basin asymmetry with their respective basins in excel, (ie DB1=0.4, DB2=0.6 etc) for further morphometric calculations.
I'll appreciate your kind help on this.

Fred
